planning/scenario_planning/common/motion_velocity_optimizer/src/motion_velocity_optimizer.cpp

```diff
@@ -661,18 +708,77 @@ void MotionVelocityOptimizer::updateExternalVelocityLimit(const double dt)
   }
 }
 
-void MotionVelocityOptimizer::timerCallback(const ros::TimerEvent & e)
+void MotionVelocityOptimizer::timerCallback()
 {
-  const double dt = (e.current_expected - e.last_expected).toSec();
+  const double dt = 1.0 / external_velocity_limit_update_rate_;
   updateExternalVelocityLimit(dt);
 }
 
-int main(int argc, char ** argv)
+void MotionVelocityOptimizer::blockUntilVehiclePositionAvailable(const tf2::Duration & timeout)
 {
-  ros::init(argc, argv, "motion_velocity_optimizer");
-  MotionVelocityOptimizer obj;
+  auto cti = std::make_shared<tf2_ros::CreateTimerROS>(
+    this->get_node_base_interface(), this->get_node_timers_interface());
+  tf_buffer_->setCreateTimerInterface(cti);
+
+  while (rclcpp::ok()) {
+    static constexpr auto input = "map", output = "base_link";
+    auto tf_future = tf_buffer_->waitForTransform(
+      input, output, tf2::TimePointZero, tf2::durationFromSec(0.0), [](auto &) {});
+    const auto status = tf_future.wait_for(timeout);
+    if (status == std::future_status::ready) {
+      break;
+    } else {
+      RCLCPP_INFO(
+        get_logger(), "waiting another %d seconds for %s->%s transform",
+        std::chrono::duration_cast<std::chrono::seconds>(timeout).count(), input, output);
+    }
+  }
+}
 
-  ros::spin();
+rcl_interfaces::msg::SetParametersResult MotionVelocityOptimizer::paramCallback(
+  const std::vector<rclcpp::Parameter> & parameters)
+{
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = true;
+  result.reason = "success";
+
+  try {
+    UPDATE_PARAM(max_velocity);
```

Comment:
This doesn't have the strong exception safety. If one update fails, the previous one succeed. Can you live with that? If you want to support only the case with one parameter changed at a time, that's ok

Reply:
Thank you. At least there is no option to change several parameters simultaneously, and this is only for the parameter tuning, so this is enough I think.

Reply:
Good with me